src/bors/handlers/review.rs

```diff
@@ -85,18 +78,10 @@ pub(super) async fn handle_push_to_pull_request(
     payload: PullRequestPushed,
 ) -> anyhow::Result<()> {
     let pr = &payload.pull_request;
-    let pr_model = db
-        .get_or_create_pull_request(repo_state.repository(), pr.number)
-        .await?;
-
-    if !pr_model.is_approved() {
-        return Ok(());
-    }
-
-    let pr_number = pr_model.number;
-    db.unapprove(repo_state.repository(), pr_number).await?;
-    handle_label_trigger(&repo_state, pr_number, LabelTrigger::Unapproved).await?;
-    notify_of_pushed_pr(&repo_state, pr_number, pr.head.sha.clone()).await
+    unapprove_if_approved(&repo_state, &db, pr, || async {
```

Comment:
I meant just the shared part before the last line. Can't we just do
```rust
unapprove_if_approved(&repo_state, &db, pr).await?;
notify_of_pushed_pr(&repo_state, pr.number, pr.head.sha.clone()).await
```
? :sweat_smile:

Reply:
If we do that it will still post a comment event if this PR is not approved 😅

Reply:
Right, so what about returning `Result<bool>`?

Reply:
Hmmm, I'd prefer we kept the original version. IMO condition function should be pure, so I'm not really comfortable with returning `Result<bool>`

Reply:
I'm fine with going back to the original, but I'm intrigued by your last sentence :) What's a "condition function" and how does this interact with pure functions? Pretty sure no functions in this file are pure :D

Reply:
Ah, I see. The function is already called `unapprove_if_approved`, so it seems natural to me that it signals to the caller if it actually did perform the unapproval or not. But up to you.